compound.hh

```diff
@@ -117,15 +118,15 @@ public:
     managed_bytes serialize_optionals(std::span<const bytes_opt> values) const {
         return serialize_value(boost::make_iterator_range(values.begin(), values.end()) | boost::adaptors::transformed([] (const bytes_opt& bo) -> bytes_view {
             if (!bo) {
-                throw std::logic_error("attempted to create key component from empty optional");
+                throw exceptions::invalid_request_exception("Invalid null value in condition for column {}");
```

Comment:
You can include the name of the column, but for that you need the schema and the index of the column with the null value. Given that, you can look up the column with the given index in the partition or clustering key.

Reply:
I just saw now that this is in `compound.hh`. I think this file is too deep in the code to throw `invalid_request_exception()` from here. This code can be used in other contexts, not just when fulfilling a request.
I think the way to go is to patch front-ends like CQL or alternator to check the optionals themselves and report any nulls themselves before attempting to create a key.

Reply:
I agree with Botond. I think the caller should check these optionals and throw whatever they want before calling this function - the logic_error in this function should remain as a guard against bugs in the caller.

As Botond said, the CQL code that calls this (e.g., statement_restrictions.cc uses from_optional_exploded() which calls serialize_optionals()) should check the optional before the call and throw the CQL exception `exceptions::invalid_request_exception`. If there was a use of this function in Alternator (I believe there isn't), it would throw a different exception - Alternator's `api_error::validation`.